evergreen/Dockerfile-al2-aarch64

```diff
@@ -0,0 +1,30 @@
+# This build environment follows the one maintained by SRE team in the Envoy RPM spec:
+# https://github.com/xgen-cloud/rpm-spec/blob/master/build/envoy/SPECS/envoy.spec
+FROM amazonlinux:2
+
+RUN amazon-linux-extras install epel
+RUN yum groupinstall -y 'Development Tools'
+RUN yum install -y clang python3
```

Comment:
AL2 uses the system clang, which is Clang 11. When we update the Envoy version, we need Clang 12. One option is to use MongoDB toolchain. I think SRE repo does that. I'll leave that to a future work.

Reply:
Yes, I think it will be more stable in general if we can move to the MongoDB toolchain.

Reply:
When I upgraded the Envoy submodule in serverless-proxy to 1.24, it looks like it now expects Clang 14. Just something to keep in mind for future work 😄